File: domain-driven-design/go/service/order_service.go
package service

import (
  "fmt"
  "go-order-system/domain"
  "go-order-system/repository"
)

// 创建订单
func CreateOrderService(id int, customerName string, amount float64) {
  order := domain.NewOrder(id, customerName, amount)
  if order != nil {
    repository.SaveOrder(order)
    fmt.Printf("订单 ID %d 创建成功\n", id)
  }
}

// 取消订单
func CancelOrderService(id int) {
  order := repository.GetOrderById(id)
  if order != nil {
    order.Cancel()
  } else {
    fmt.Printf("未找到 ID %d\n", id)
  }
}

// 查询订单
func QueryOrderService(id int) {
  order := repository.GetOrderById(id)
  if order != nil {
    order.Display()
  } else {
    fmt.Printf("未找到 ID %d\n", id)
  }
}

File: domain-driven-design/c/service/order_service.c
// 应用服务层（Application Layer） - order_service.c
#include "../service/order_service.h"
#include "../repository/order_repository.h"
#include <stdio.h>

// 创建订单
void createOrderService(int id, const char *customerName, double amount)
{
    Order *order = createOrder(id, customerName, amount);
    if (order)
    {
        saveOrder(order);
        printf("订单 ID %d 创建成功\n", id);
    }
}

// 取消订单
void cancelOrderService(int id)
{
    Order *order = getOrderById(id);
    if (order)
    {
        cancelOrder(order);
    }
    else
    {
        printf("未找到 ID %d\n", id);
    }
}

// 查询订单
void queryOrderService(int id)
{
    Order *order = getOrderById(id);
    if (order)
    {
        displayOrderInfo(order);
    }
    else
    {
        printf("未找到 ID %d\n", id);
    }
}
